src/main.rs

```diff
@@ -178,16 +180,45 @@ fn start_zenith(
         };
 
         debug!("Create Renderer");
-        let mut r = TerminalRenderer::new(
-            rate,
-            cpu_height as i16,
-            net_height as i16,
-            disk_height as i16,
-            process_height as i16,
-            sensor_height as i16,
-            graphics_height as i16,
-            db,
-        );
+
+        let mut geometry: Vec<(Section, f64)> = Vec::new();
```

Comment:
Could you refactor this section into a function? If possible, I'd also like the parameter checking to be done in main before `start_zenith` is called.